lib/game.rb

```diff
@@ -0,0 +1,85 @@
+class Game
+  attr_accessor :board, :player_1, :player_2, :name
+
+  # Win combinations constant
+  WIN_COMBINATIONS = [
+    [0, 1, 2], # Top row
+    [3, 4, 5], # Middle row
+    [6, 7, 8], # Bottom row
+    [0, 3, 6], # Left column
+    [1, 4, 7], # Middle column
+    [2, 5, 8], # Right column
+    [0, 4, 8], # Left Diagonal
+    [2, 4, 6]  # Right Diagonal
+  ]
+  
+  # Game methods
+  def initialize(player_1 = Players::Human.new("X", name), player_2 = Players::Human.new("O", name), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+    self
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def won?
+    # THIS FUCKING SUCKEDDDD
+    win = false
+    WIN_COMBINATIONS.each do |combo|
+      win = combo if check_combo(combo)
+    end
+    win
+  end
+
```

Comment:
Nice job working out the logic to a difficult problem!  Good idea to abstract out the logic into two methods.

As another alternative, you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```

Reply:
Thanks! I had help from @DRBragg with abstracting the method.  I was pretty frustrated with this method though. I believe his words were "You look like you're about to punch a child in the face."